packages/cozy-scanner/src/DocumentQualification.jsx

```diff
@@ -0,0 +1,201 @@
+            <InputGroup
+              fullwidth
+              append={<span className="u-pr-1">{filename_extension}</span>}
+              className="u-bdrs-3"
+            >
+              <Input
+                placeholder={t('Scan.filename')}
+                value={filename}
+                onChange={event => {
+                  //If the user write something once, we don't want to rename the file automatically anymore
+                  if (!hasUserWrittenFileName) {
+                    this.setState({ hasUserWrittenFileName: true })
+                  }
+                  //If we left an empty value, then we reset the behavior
+                  if (event.target.value === '')
+                    this.setState({ hasUserWrittenFileName: false })
+                  this.handleFileNameChange(event.target.value)
+                }}
+                onFocus={() => {
+                  if (!hasUserWrittenFileName)
+                    this.textInput.current.setSelectionRange(0, filename.length)
+                }}
+                onBlur={() => {
+                  if (filename === '') {
+                    this.handleFileNameChange(
+                      this.getFilenameFromCategory(selected, t)
+                    )
+                  }
+                }}
+                inputRef={this.textInput}
+                id={id_filename_input}
+              />
+            </InputGroup>
+          </>
+        )}
+
+        {title && (
+          <div className="u-flex u-flex-items-center u-mt-1-half">
+            <Icon icon="qualify" />
+            <Title
+              className={classNames(
+                styles['grid-item-title'],
+                'u-pl-half u-mb-1'
+              )}
+            >
+              {title}
+            </Title>
+          </div>
+        )}
+        <Grid container spacing={1}>
+          <GridItem
+            onClick={() => this.onSelect({ categoryLabel: null, itemId: null })}
+          >
+            <CategoryGridItem
+              isSelected={selected.categoryLabel === null}
+              theme={t(`Scan.themes.undefined`)}
+            />
+          </GridItem>
+
+          {themes.map((category, i) => {
+            return (
+              <DocumentCategory
+                onSelect={item => this.onSelect(item)}
+                category={category}
+                items={getItemsByCategory(category)}
+                key={i}
+                isSelected={selected.categoryLabel === category.label}
+                selectedItem={
+                  selected.itemId ? getItemById(selected.itemId) : {}
+                }
+                t={t}
+              />
+            )
+          })}
+        </Grid>
+      </MuiCozyTheme>
+    )
+  }
+}
+
+DocumentQualification.defaultProps = {
+  editFileName: false
+}
+DocumentQualification.propTypes = {
+  /**
+   * This callback is called after a select.
+   *
+   */
+  onQualified: PropTypes.func,
```

Comment:
```suggestion
  /** Called after category selection */
  onQualified: PropTypes.func,
```

Then why isn't the name onCategorySection ;)

Reply:
Because now we can also changed the filename. Updated the comment

Reply:
Hmm, in fact, since I also have a callback for the onFileNameChange, I think I can remove the stuff related to filename and rename it to onCategorySelection 🤔